lib/Doctrine/Common/DataFixtures/Purger/ORMPurger.php

```diff
@@ -96,73 +99,80 @@ public function getObjectManager()
     /** @inheritDoc */
     public function purge()
     {
-        $classes = [];
+        $connection = $this->em->getConnection();
+
+        if ($this->cachedSqlStatements === null) {
```

Comment:
Calls to `setEntityManager` or `setPurgeMode` must invalidate the cache (at least if the new value is different from the old one

Reply:
done